### ISSUE TITLE
Question About Loss Behavior During SFT Training

### ISSUE THREAD

**Issue by GWwangshuo** [jingyaogong/minimind#566]
I trained the model using the default hyperparameters in the SFT stage, but the training loss does not appear to converge. Could there be an issue with my setup, and how should I address it?

<img width="1717" height="1010" alt="Image" src="https://github.com/user-attachments/assets/734102f3-39e7-4c8d-8bfa-2d4fa3455771" />

**Comment by jingyaogong** [comment]
- You're using `5e-7 lr × 16 bs`, 1.5k steps
- I'm using `5e-6 lr × 32 bs`, ~0.7k steps (equivalent compute)

So your learning rate is effectively 1/5 of mine. I used 5x the learning rate and only got ~0.2 loss reduction, while you got ~0.1 reduction at a lower learning rate—this ratio is actually reasonable and doesn't indicate "failure to converge."

<img width="1474" height="1252" alt="Image" src="https://github.com/user-attachments/assets/08bdaadb-6022-432a-b52c-039c37cf0b8e" />

---

Does loss matter? Of course it does, but staring at absolute values in a small-data regime is meaningless.

<img width="1696" height="1184" alt="Image" src="https://github.com/user-attachments/assets/c37a2ba8-fb9a-4b07-81c3-dd055490615d" />

Look at the LLaMA-7B training curve: going from 2.1 to 2.0—a mere 0.1 loss drop—requires approximately **70B tokens** (~350GB of data).

You think the curve looks "flat" because you're standing on the Earth's surface, measuring with a 100-meter or maybe 1000-meter ruler, and asking: "Why isn't the Earth round?"

At this data scale, this level of loss fluctuation and absolute value is meaningless.
